src/Components/App/App.tsx

```diff
@@ -22,23 +22,20 @@ function App() {
   const callApi = async () => {
     try {
       const ids: Ids = await getIDs();
-      console.log("ids:", ids);
-      const randomPieces: number[] = getRandomIds(ids);
-      console.log("randomPieces", randomPieces);
-      for (const id of randomPieces) {
+      const id: number = getRandomIds(ids);
         const data = await getArt(id);
         if (data.primaryImage && data.message !== "ObjectID not found") {
           const cleanData: any = cleanUpData(data);
           setPieces((prev) => [...prev, cleanData]);
-        }
+      } else {
+        callApi()
       }
     } catch (error: unknown) {
       if (error instanceof Error && error.message.includes("Server error")) {
```

Comment:
This looks reasonable. But I am curious if the if statement on line 34 is entirely necessary.  Were you seeing it hit the catch block for errors you didn't want to take this action for?